docs/source/tutorial_crowdsourcing.md

```diff
@@ -0,0 +1,252 @@
+
+### Sample Task: Collecting Data
+
+One of the biggest use cases of Mechanical Turk is to collect natural
+language data from human Turkers.
+
+As an example, the [QA data collection
+task](https://github.com/facebookresearch/ParlAI/blob/master/parlai/crowdsourcing/tasks/qa_data_collection/)
+does the following:
+
+1.  Pick a random Wikipedia paragraph from the SQuAD dataset.
+2.  Ask a Turker to provide a question given the paragraph.
+3.  Ask the same Turker to provide an answer to their question.
+
+In `QADataCollectionWorld`, there are two agents: one is the human
+Turker (`Agent`) and the other is the task agent (`SquadQATeacher`
+from SQuAD) that provides the Wikipedia paragraph.
+
+`QADataCollectionWorld` uses `.question` and `.answer` attributes to denote what stage the
+conversation is at. The task lasts for two *turns*: each turn means that `world.parley()` has been
+called once. After two turns, the task is finished, and the Turker's work is
+submitted for review.
+
+Creating Your Own Task
+----------------------
+
+Mephisto provides a generic MTurk dialog interface that you can use to
+implement any kind of dialog task. To create your own task, start by
+reading the README of the existing task in [`parlai/crowdsourcing/`](https://github.com/facebookresearch/ParlAI/tree/master/parlai/crowdsourcing) that your desired task most resembles, and then subclass the appropriate components in order to write your own task. You may need to subclass the following classes (or classes that inherit from them):
+
+- `CrowdOnboardingWorld`: the base world class that handles the logic for determining whether a crowdsource worker passes or fails onboarding. This logic is typically used to judge whether the crowdsource worker is likely to be able to complete your task in an acceptable manner.
+- `CrowdTaskWorld`: the base world class that handles the logic for how each of the human or bot agents will act during each turn of the task, as defined by a call to `CrowdTaskWorld.parley()`.
+- `Blueprint`: the base class containing task-specific logic for setting up a task run. See the Mephisto [Blueprint README](https://github.com/facebookresearch/Mephisto/blob/master/mephisto/abstractions/blueprints/README.md) and [architecture overview](https://github.com/facebookresearch/Mephisto/blob/master/docs/architecture_overview.md) for more information.
+- `BlueprintArgs`: defines the specific arguments needed for configuring a blueprint. Subclasses of `BlueprintArgs` retain all arguments defined by their superclasses and add additional task-specific ones.
+
+If you are creating a new `Blueprint`, you will currently need to create a new `run.py` file in which to call your `Blueprint` so that its arguments can be read in correctly by [Hydra](https://github.com/facebookresearch/hydra); this should no longer be necessary as of the upcoming Hydra 1.1. You will likely need to create the following helper files for your task:
```

Comment:
As an aside, in the future, people will still want to create their own `run.py` files to have more control over setup, or being able to initialize `SharedParlAITaskState`-based arguments. I'm not sure that this quite captures the duality of those, but for now it's fine.

Reply:
Yeah, good point - I've just added these as other use cases for creating a new run.py file